src/main/java/games/strategy/util/Version.java

```diff
@@ -131,42 +135,41 @@ public int hashCode() {
   }
 
   @Override
-  public int compareTo(final Version o) {
-    return o == null ? -1 : compareTo(o, false);
+  public int compareTo(final Version other) {
+    checkNotNull(other);
+
+    return compareTo(other, false);
   }
 
   private int compareTo(final Version other, final boolean ignoreMicro) {
-    if (other == null) {
-      return -1;
-    }
-    if (other.m_major > m_major) {
+    if (m_major > other.m_major) {
       return 1;
-    }
-    if (other.m_major < m_major) {
+    } else if (m_major < other.m_major) {
       return -1;
-    } else if (other.m_minor > m_minor) {
+    } else if (m_minor > other.m_minor) {
       return 1;
-    } else if (other.m_minor < m_minor) {
+    } else if (m_minor < other.m_minor) {
       return -1;
-    } else if (other.m_point > m_point) {
+    } else if (m_point > other.m_point) {
       return 1;
-    } else if (other.m_point < m_point) {
+    } else if (m_point < other.m_point) {
       return -1;
     } else if (!ignoreMicro) {
-      if (other.m_micro > m_micro) {
+      if (m_micro > other.m_micro) {
         return 1;
-      } else if (other.m_micro < m_micro) {
+      } else if (m_micro < other.m_micro) {
         return -1;
       }
     }
-    // if the only difference is m_micro, then ignore
     return 0;
```

Comment:
Without the overload (and even with the ignoreMicro overload, but not that nicely and without being able to merge those 2 methods), we can replace those annoying if else statements with 4 nice lambda expressions using
```
Comparator.comparingInt(Version::getMajor)
  .thenComparingInt(Version::getMinor)
  .thenComparingInt(Version::getPoint)
  .thenComparingInt(Version::getMicro)
  .compare(this, other);
```
If we want to keep the overload the last lambda expression needs to be modified accordingly so the values passed to method are equal when ignoreMicro is true.

Reply:
Definitely and improvement for readability.  I'll try adopting that pattern.